scripts/source-maintenance.sh

```diff
@@ -40,7 +46,24 @@ fi
 COPYRIGHT_YEARS=`date +"1996-%Y"`
 echo "s/1996-2[0-9]+ The Squid Software Foundation and contributors/${COPYRIGHT_YEARS} The Squid Software Foundation and contributors/g" >>boilerplate_fix.sed
 
-srcformat ()
+updateIfChanged ()
+{
+	original="$1"
+	updated="$2"
+	message="$3"
+
+	test -e "${original}" -a -e "${updated}" || return
+	md51=`cat "${original}" | ${MD5}`;
+	md52=`cat "${updated}" | ${MD5}`;
+	if test "${md51}" != "${md52}" ; then
+		echo "NOTICE: File ${original} changed: ${message}"
+		mv "${updated}" "${original}" || return
+	else
+		rm -f "${updated}"
```

Comment:
The case you are worried about is one which indicated a major fault in the filesystem and security permissions. Modifying the suggestion to use exit rather than return.

Reply:
I consider this change request addressed, thank you.

> The case you are worried about is one which indicated a major fault in the filesystem and security permissions.

... or just the same developer running `sudo scripts/source-maintenance.sh` in another window/session. Or a plugin script creating its own output file with the wrong permissions. Etc... 

I have no energy to object to a sudden and partial classification of command errors as "deserving an immediate death inside a low-level function" even though I consider such classification an overreaction and do not like that its implementation leaves the code reader guessing.